tests/components/acmeda/test_config_flow.py

```diff
@@ -0,0 +1,143 @@
+"""Define tests for the Acmeda config flow."""
+import aiopulse
+from asynctest.mock import patch
+import pytest
+
+from homeassistant import data_entry_flow
+from homeassistant.components.acmeda.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_HOST
+
+from tests.common import MockConfigEntry
+
+DUMMY_HOST1 = "127.0.0.1"
+DUMMY_HOST2 = "127.0.0.2"
+
+CONFIG = {
+    CONF_HOST: DUMMY_HOST1,
+}
+
+
+@pytest.fixture
+def mock_hub_discover():
+    """Mock the hub discover method."""
+    with patch("aiopulse.Hub.discover") as mock_discover:
+        yield mock_discover
+
+
+@pytest.fixture
+def mock_hub_run():
+    """Mock the hub run method."""
+    with patch("aiopulse.Hub.run") as mock_run:
+        yield mock_run
+
+
+async def async_generator(items):
+    """Async yields items provided in a list."""
+    for item in items:
+        yield item
+
+
+async def test_show_form_no_hubs(hass, mock_hub_discover):
+    """Test that flow aborts if no hubs are discovered."""
+    mock_hub_discover.return_value = async_generator([])
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "all_configured"
+
+    # Check we performed the discovery
+    assert len(mock_hub_discover.mock_calls) == 1
+
+
+async def test_show_form_one_hub(hass, mock_hub_discover, mock_hub_run):
+    """Test that a config is created when one hub discovered."""
+
+    dummy_hub_1 = aiopulse.Hub(DUMMY_HOST1)
+    dummy_hub_1.id = "ABC123"
+
+    mock_hub_discover.return_value = async_generator([dummy_hub_1])
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == dummy_hub_1.id
+    assert result["result"].data == {
+        "host": DUMMY_HOST1,
+    }
+
+    # Check we performed the discovery
+    assert len(mock_hub_discover.mock_calls) == 1
+
+
+async def test_show_form_two_hubs(hass, mock_hub_discover):
+    """Test that the form is served when more than one hub discovered."""
+
+    dummy_hub_1 = aiopulse.Hub(DUMMY_HOST1)
+    dummy_hub_1.id = "ABC123"
+
+    dummy_hub_2 = aiopulse.Hub(DUMMY_HOST1)
+    dummy_hub_2.id = "DEF456"
+
+    mock_hub_discover.return_value = async_generator([dummy_hub_1, dummy_hub_2])
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)